python/pyspark/sql.py

```diff
@@ -2136,9 +2136,9 @@ def __getitem__(self, item):
 
     def __getattr__(self, name):
         """ Return the column by given name """
-        if isinstance(name, basestring):
```

Comment:
I don't think we need this check because `getattr` can only be called with strings:

```python
>>> foo = []
>>> getattr(foo, 123)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: getattr(): attribute name must be string
```